File: Medium/Python/SortingAndSearching/SearchMatrix.py
class Solution(object):
    def searchMatrix(self, matrix, target):
        """
            :type matrix: List[List[int]]
            :type target: int
            :rtype: bool
            """
        if len(matrix) == 0 or len(matrix[0]) == 0:
            return False
        if target < matrix[0][0] or target > matrix[-1][-1]:
            return False
        for i in xrange(len(matrix)):
            try:
                matrix[i].index(target)
                return True
            except:
                x = 'nothing'
        return False

File: Medium/Java/SortAndSearch/SearchMatrix.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        if (matrix == null || matrix.length < 1 || matrix[0].length < 1)
            return false;
        int col = 0, row = matrix.length - 1;
        while (col < matrix[0].length && row >= 0) {
            if (matrix[row][col] > target)
                row --;
            else if (matrix[row][col] < target)
                col ++;
            else return true;
        }
        return false;
    }
}
